idaes_examples/notebooks/docs/unit_models/custom_unit_models/Aq_property.py

```diff
@@ -0,0 +1,306 @@
+
+        init_log.info("Initialization Complete.")
+
+    def release_state(self, flags, outlvl=idaeslog.NOTSET):
+        """
+        Method to release state variables fixed during initialization.
+
+        Keyword Arguments:
+            flags : dict containing information of which state variables
+                    were fixed during initialization, and should now be
+                    unfixed. This dict is returned by initialize if
+                    hold_state=True.
+            outlvl : sets output level of logging
+        """
+        init_log = idaeslog.getInitLogger(self.name, outlvl, tag="properties")
+
+        if flags is None:
+            return
+        # Unfix state variables
+        revert_state_vars(self, flags)
+        init_log.info("State Released.")
+
+
+@declare_process_block_class("AqPhaseStateBlock", block_class=_StateBlock)
+class AqPhaseStateBlockData(StateBlockData):
+    """
+    An example property package for ideal gas properties with Gibbs energy
+    """
+
+    def build(self):
+        """
+        Callable method for Block construction
+        """
+        super(AqPhaseStateBlockData, self).build()
+        self._make_state_vars()
+
+    def _make_state_vars(self):
+        self.flow_vol = Var(
+            initialize=1,
+            domain=NonNegativeReals,
+            doc="Total volumetric flowrate",
+            units=units.ml / units.min,
+        )
+
+        salts_conc = {"NaCl": 0.15, "KNO3": 0.2, "CaSO4": 0.1}
+        self.conc_mass_comp = Var(
+            salts_conc.keys(),
+            domain=NonNegativeReals,
+            initialize=1,
+            doc="Component mass concentrations",
+            units=units.g / units.kg,
+        )
+        self.pressure = Var(
+            domain=NonNegativeReals,
+            initialize=1,
+            bounds=(1, 5),
+            units=units.atm,
+            doc="State pressure [atm]",
+        )
+
+        self.temperature = Var(
+            domain=NonNegativeReals,
+            initialize=300,
+            bounds=(273, 373),
+            units=units.K,
+            doc="State temperature [K]",
+        )
+
+        def material_flow_expression(self, j):
+            if j == "H2O":
+                return self.flow_vol * self.params.dens_mass
+            else:
+                return self.flow_vol * self.conc_mass_comp[j]
+
+        self.material_flow_expression = Expression(
+            self.component_list,
+            rule=material_flow_expression,
+            doc="Material flow terms",
+        )
+
+        def enthalpy_flow_expression(self):
+            return (
+                self.flow_vol
+                * self.params.dens_mass
+                * self.params.cp_mass
+                * (self.temperature - self.params.temperature_ref)
+            )
+
+        self.enthalpy_flow_expression = Expression(
+            rule=enthalpy_flow_expression, doc="Enthalpy flow term"
+        )
+
+    def get_mass_comp(self, j):
```

Comment:
These appear to be non-standard methods. What do you intended to use them for?

Reply:
I intend to use this for getting mass of the component in the phase, this will be used in the mass transfer constraint in the liquid_extraction.py file.

Reply:
You should not do that, as no other property package supports this method. The unit model can directly access the property anyway, so I cannot see why this would be needed.